File: src/main/java/hacker/solution/TypeSize.java
package hacker.solution;

import java.util.Scanner;

public class TypeSize {
    public static void fitIn(String s) {
        try {
            Long l = Long.parseLong(s);
            System.out.printf("%d can be fitted in:\n", l);
            if (l >= -Math.pow(2, 7) && l < Math.pow(2, 7)) System.out.println("* byte");
            if (l >= -Math.pow(2, 15) && l < Math.pow(2, 15)) System.out.println("* short");
            if (l >= -Math.pow(2, 31) && l < Math.pow(2, 31)) System.out.println("* int");
            System.out.println("* long");
        } catch (NumberFormatException e) {
            System.out.printf("%s can't be fitted anywhere.\n", s);
        }
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String s1 = sc.nextLine();
        int j = 0;
        try {
            j = Integer.parseInt(s1);
        } catch (NumberFormatException e) {
            System.exit(0);
        }
        for (int i = 0; i < j; i++) {
            String s = sc.nextLine();
            fitIn(s);
        }
        sc.close();
    }
}

File: src/main/kotlin/hacker/solutionk/TypeSize.kt
package hacker.solutionk

import java.util.*

fun fitIn(s: String) {
    try {
        val l = java.lang.Long.parseLong(s)
        System.out.printf("%d can be fitted in:\n", l)
        try {
            java.lang.Byte.parseByte(s);
            println("* byte")
        } catch (e1: NumberFormatException) {}
        try {
            java.lang.Short.parseShort(s);
            println("* short")
        } catch (e1: NumberFormatException) {}
        try {
            java.lang.Integer.parseInt(s);
            println("* int")
        } catch (e1: NumberFormatException) {}
        println("* long")
    } catch (e: NumberFormatException) {
        System.out.printf("%s can't be fitted anywhere.\n", s)
    }

}

fun main(args: Array<String>) {
    val sc = Scanner(System.`in`)
    val s1 = sc.nextLine()
    var j = 0
    try {
        j = Integer.parseInt(s1)
    } catch (e: NumberFormatException) {
        System.exit(0)
    }

    for (i in 0 until j) {
        val s = sc.nextLine()
        fitIn(s)
    }
    sc.close()
}
